test/TestProjects/MgmtPartialResource/Generated/PartialVmssResource.cs

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager;
+
+namespace MgmtPartialResource
+{
+    /// <summary>
+    /// A class extending from the VirtualMachineScaleSetResource in MgmtPartialResource along with the instance operations that can be performed on it.
+    /// You can only construct a <see cref="PartialVmssResource" /> from a <see cref="ResourceIdentifier" /> with a resource type of Microsoft.Compute/virtualMachineScaleSets.
+    /// </summary>
+    public partial class PartialVmssResource : ArmResource
+    {
+        /// <summary> Generate the resource identifier of a <see cref="PartialVmssResource"/> instance. </summary>
+        internal static ResourceIdentifier CreateResourceIdentifier(string subscriptionId, string resourceGroupName, string virtualMachineScaleSetName)
```

Comment:
I'd like to know why this is internal. The internal here means if we want to get a resource identifier, we need to use the corresponding `CreateResourceIdentifier` method in VmssResource of compute, which will result in that we need to add the package `Azure.ResourceManager.Compute` as an reference, or we need to directly use the ctor of `ResourceIdentifier`. And I think both of the options are not friendly for the user.

Reply:
Another finding that makes me think maybe we should change this method to public method: In Consumption we have a partial resource called [BillingAccountConsumptionResource](https://github.com/Azure/azure-sdk-for-net/blob/7d344e2ccdd6626a1dce393240027480527696bd/sdk/consumption/Azure.ResourceManager.Consumption/src/Customized/BillingAccountConsumptionResource.cs#L22), but unfortunately we don't have a BillingAccountResource in the RP `Billing`, in this case it will be difficult for the customer to get the correct resource identifier.